docs/docs/guides/smart_contracts/testing_contracts/index.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Testing Contracts
+---
+
+Aztec contracts can be tested in a variety of ways depending on the needs of a particular application and the complexity of the interactions they must support. 
+
+## Pure Noir tests
+
+Noir supports the `#[test]` annotation which can be used to write simple logic tests on isolated utility functions. These tests only make assertions on algorithms and cannot interact with protocol-specific constructs such as `storage` or `context`, but are extremely fast and can be useful in certain scenarios. 
+
+#include_code pure_noir_testing /noir-projects/noir-contracts/contracts/card_game_contract/src/cards.nr rust
+
+To learn more about Noir testing, please refer to [the docs](https://Noir-lang.org/docs/tooling/testing/)
+
+## TXE
+
+In order to interact with the protocol, aztec contracts leverage the power of oracles: functions that reach out to the outside world and are able to query and manipulate data outside of itself. The values returned by oracles are then constrained inside Noir and the modifications to the blockchain state later verified to adhere to the protocol rules by our kernel circuits.
+
+However, all of this is often not necessary to ensure the contract logic itself is sound, and all that we need is an entity to provide values consistent with real execution. This is where our TXE (Testing eXecution Environment) comes in!
+
+TXE is a JSON RPC server much like PXE, but provides an extra set of oracle functions called `cheatcodes` that allow developers to manipulate the state of the chain and simulate contract execution. Since TXE skips most of the checks, blockbuilding and other intrincacies of the aztec protocol, it is much faster to run than simulating everything in the sandbox.
+
+### Running TXE
+
+In order to use the TXE, it must be running on a known address. Assuming the default `http://localhost:8080`, contract tests would be run with:
+
+ `nargo test --oracle-resolver http://localhost:8080`
```

Comment:
Will this be `aztec-nargo test --oracle-resolver http://localhost:8080` when external developers are using it?

Reply:
Yes, but I intend to create a cleaner command with the CLI (probably `aztec test`) that does all the heavy lifting

Reply:
ok. should we update it to say `aztec-nargo`? or wait until we have a cleaner command?

Reply:
Alright! `aztec-nargo` should work for the time being.